src/lib/logical_query_plan/union_node.cpp

```diff
@@ -45,6 +45,27 @@ bool UnionNode::is_column_nullable(const ColumnID column_id) const {
   return left_input()->is_column_nullable(column_id) || right_input()->is_column_nullable(column_id);
 }
 
+std::shared_ptr<LQPUniqueConstraints> UnionNode::unique_constraints() const {
+  switch (set_operation_mode) {
+    case SetOperationMode::Positions:
+      // UnionPositions merges two reference tables with the same original table(s), computing a Set Union of their
+      // pos lists. Therefore, unique constraints remain valid.
```

Comment:
... In the cases that we have seen so far (usually disjunctions turned into individual nodes by the PredicateSplitUpRule), the constraints have remained the same on the left and right sides. This is not necessarily true in the future. We assert this behavior so that we are aware if this ever changes.